kasa/discover.py

```diff
@@ -193,19 +193,21 @@ async def discover(
         return protocol.discovered_devices
 
     @staticmethod
-    async def discover_single(host: str) -> SmartDevice:
+    async def discover_single(
+        host: str, *, port=TPLinkSmartHomeProtocol.DEFAULT_PORT
```

Comment:
Let's leave the `port` as `None` here and leave it to the protocol to handle. This makes the diff much smaller and leaves the default handling to the implementation class.

The reason for this is that there is an open PR that adds handling for kasa protocol using different encryption scheme (#471), so I think it's for the best to keep this change as simple and contained as possible in case as it separates this piece of implementation detail.

Reply:
Hi there, I think you are referring to my PR #477 here?  My 2 cents here for this change, particularly as it's going to be impacted if the protocols supported goes above 1, is that it might be better to make it more specific to the use case of port forwarding, i.e pass in a list of --portforward params, maybe looking like 9999-1234,20002-12345,80-81 or similar (klap protocol uses 20002 and 80).  These can then be mapped within each protocol if the mapping matches their expected port.  The use case seems to presume that the person configuring the network knows the correct port for the protocol when they setup the forwarding rule so shouldn't be a problem just passing the forwarding rule itself.

Reply:
Ah, indeed. I haven't given much thought on that yet, but nevertheless, I think it makes sense to keep the diff small and avoid importing `TPLinkSmartHomeProtocol` only for the constant through all device implementation classes.

Reply:
Thank you for the suggestion. I updated PR.